src/plugins/vis_type_timeseries/server/lib/timeseries_visualization_client_wrapper.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SavedObjectsClientContract,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsClientWrapperOptions,
+  SavedObjectsCreateOptions,
+} from '../../../../core/server';
+import { getDataSourceEnabled } from './services';
+
+export const TIMESERIES_VISUALIZATION_CLIENT_WRAPPER_ID = 'timeseries-visualization-client-wrapper';
+
+export const timeSeriesVisualizationClientWrapper: SavedObjectsClientWrapperFactory = (
+  wrapperOptions: SavedObjectsClientWrapperOptions
+) => {
+  const createForTimeSeries = async <T = unknown>(
+    type: string,
+    attributes: T,
+    options?: SavedObjectsCreateOptions
+  ) => {
+    if (type !== 'visualization' || !getDataSourceEnabled().enabled) {
+      return await wrapperOptions.client.create(type, attributes, options);
+    }
+
+    // @ts-expect-error
+    const visState = JSON.parse(attributes.visState);
+
+    if (visState.type !== 'metrics' || !visState.params) {
+      return await wrapperOptions.client.create(type, attributes, options);
+    }
+
+    const newReferences = options?.references?.filter(
+      (reference) => reference.type !== 'data-source'
+    );
+
+    if (visState.params.data_source_id) {
+      if (await checkIfDataSourceExists(visState.params.data_source_id, wrapperOptions.client)) {
```

Comment:
how do we handle network failure when check data source exists? Will creation be successful?

Reply:
Added `try catch` block to throw an error in the case that the promise rejects.